### PR TITLE
Arnej/create filter in intermediates

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst @toregge @havardpe please review

### CODE REVIEW

File: vespalib/src/vespa/vespalib/regex/regex.h [hunk 1]
```diff
@@ -44,7 +44,8 @@ class Regex {
     // TODO consider using type-safe parameter instead.
     enum Options {
         None       = 0,
-        IgnoreCase = 1
+        IgnoreCase = 1,
+        MultiLine  = 2
```
Comment (vekterli) [inline vespalib/src/vespa/vespalib/regex/regex.h:48]: According to the RE2 docs, `set_dot_nl(bool)` has the behavior "dot matches everything including new line". Maybe we should call this `DotMatchesNewline` instead to be explicit that this does not trigger any changes to how anchors work etc? `MultiLine` might imply some different semantics than what is currently in place.

**Review by vekterli — APPROVED** [review]
One comment for consideration for the regex changes. The rest of the code I leave for people with more knowledge than me 🙂

File: searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp [hunk 2]
```diff
@@ -320,6 +363,19 @@ OrBlueprint::createIntermediateSearch(const MultiSearch::Children &subSearches,
     return SearchIterator::UP(OrSearch::create(subSearches, strict, unpackInfo));
 }
 
+SearchIterator::UP
+OrBlueprint::createFilterSearch(bool strict, FilterConstraint constraint) const
+{
+    MultiSearch::Children subSearches;
+    subSearches.reserve(childCnt());
+    for (size_t i = 0; i < childCnt(); ++i) {
+        auto search = getChild(i).createFilterSearch(strict, constraint);
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp:372]: consider using inheritStrict for consistency

**Review by havardpe — APPROVED** [review]

File: searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp [hunk 3]
```diff
@@ -168,6 +168,33 @@ AndNotBlueprint::createIntermediateSearch(const MultiSearch::Children &subSearch
...
+        return Blueprint::FilterConstraint::UPPER_BOUND;
+    }
+    abort();
+}
+} // namespace <unnamed>
+
+SearchIterator::UP
+AndNotBlueprint::createFilterSearch(bool strict, FilterConstraint constraint) const
+{
+    MultiSearch::Children subSearches;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp:186]: Consider snake_case for consistency (e.g. child_strict is used further down).

File: searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp [hunk 4]
```diff
@@ -243,6 +270,22 @@ AndBlueprint::createIntermediateSearch(const MultiSearch::Children &subSearches,
     return SearchIterator::UP(search);
 }
 
+SearchIterator::UP
+AndBlueprint::createFilterSearch(bool strict, FilterConstraint constraint) const
+{
+    MultiSearch::Children subSearches;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp:276]: Consider snake_case for consistency.

File: searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp [hunk 5]
```diff
@@ -243,6 +270,22 @@ AndBlueprint::createIntermediateSearch(const MultiSearch::Children &subSearches,
...
+    subSearches.reserve(childCnt());
+    for (size_t i = 0; i < childCnt(); ++i) {
+        bool child_strict = strict && inheritStrict(i);
+        auto search = getChild(i).createFilterSearch(child_strict, constraint);
+        subSearches.push_back(search.release());
+    }
+    UnpackInfo unpackInfo;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.cpp:283]: Consider snake_case for consistency.

**Review by geirst — APPROVED** [review]
👍

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**